applications/train_with_xarray.py

```diff
@@ -0,0 +1,482 @@
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from collections import defaultdict
+import pandas as pd
+import numpy as np
+import torch.fft
+import subprocess
+import logging
+import random
+import shutil
+import psutil
+import scipy
+import torch
+import yaml
+import time
+import tqdm
+import sys
+import gc
+
+from holodecml.data import PickleReader, UpsamplingReader, XarrayReader
+from holodecml.propagation import InferencePropagator
+from holodecml.transforms import LoadTransformations
+from holodecml.models import load_model
+from holodecml.losses import load_loss
+
+import os
+import warnings
+warnings.filterwarnings("ignore")
+
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+
+
+available_ncpus = len(psutil.Process().cpu_affinity())
+
+
+# ### Set seeds for reproducibility
+def seed_everything(seed=1234):
+    random.seed(seed)
+    os.environ['PYTHONHASHSEED'] = str(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    if torch.cuda.is_available():
+        torch.cuda.manual_seed(seed)
+        torch.backends.cudnn.benchmark = True
+        torch.backends.cudnn.deterministic = True
+
+
+def dice(true, pred, k=1):
+    true = np.array(true)
+    pred = np.array(pred)
+    intersection = np.sum(pred[true == k]) * 2.0
+    dice = intersection / (np.sum(pred) + np.sum(true) + 1e-12)
```

Comment:
For small values to prevent divide by zero errors, it would be good to do a little test or check to compare adding an eps value like 1e-12 to losses vs using np.maximum(denominator, eps).

Reply:
Testing this now -- this happens a lot!